src/Ledgers.jl

```diff
@@ -12,94 +12,119 @@ Licensed under MIT License, see LICENSE.md
+using Assets: USD
 
-export Credit, Debit, Account, Ledger, Entry, AccountId, AccountInfo
-export id, balance, credit!, debit!, post!, instrument, symbol, amount, name, currency
+export Account, Ledger, Entry, AccountId, AccountCode, AccountInfo, AccountGroup
+export id, balance, credit!, debit!, post!, instrument, symbol, amount, code, name, currency
+export parent, subaccounts, subgroups
 
 abstract type Identifier end
+
 struct AccountId <: Identifier
     value::UUID
 end
+
 AccountId() = AccountId(uuid4())
 
 struct AccountCode
     value::String
 end
 
-abstract type AccountType end
-struct Credit <: AccountType end
-struct Debit <: AccountType end
+abstract type AccountType{B <: Position} end
 
-mutable struct Account{P<:Position}
+mutable struct Account{B <: Position} <: AccountType{B}
     id::AccountId
-    balance::P
+    balance::B
 end
-Account(balance::Position) = Account(AccountId(), balance)
 
-abstract type AccountNode end
-struct AccountInfo{AT<:AccountType,A<:Account} <: AccountNode
-    account::A
+Account(balance::Position) = Account{typeof(balance)}(AccountId(), balance)
+
+abstract type AccountNode{B <: Position} <: AccountType{B} end
+
+struct AccountInfo{B <: Position} <: AccountNode{B}
+    account::Account{B}
     code::AccountCode
     name::String
+    isdebit::Bool
 
-    function AccountInfo(::Type{T}, account, code, name) where {T<:AccountType}
-        return new{T}(account, code, name)
+    function AccountInfo{B}(account::Account{B}, code, name, isdebit=true, parent=nothing) where {B <: Position}
+        acc = new{B}(account, code, name, isdebit)
+        parent === nothing || push!(parent.subaccounts, acc)
+        acc
     end
 end
 
-struct AccountGroup{AT<:AccountType} <: AccountNode
+AccountInfo(account::Account{B}, code, name, isdebit=true, parent=nothing) where {B <: Position} =
+    AccountInfo{B}(account, code, name, isdebit, parent)
+
+struct AccountGroup{B <: Position} <: AccountNode{B}
+    id::AccountId
     code::AccountCode
     name::String
-    parent::Union{Nothing,AccountGroup{<:AccountType}}
-    subaccounts::Vector{AccountInfo}
-    subgroups::Vector{AccountGroup}
-
-    function AccountGroup(::Type{T}, account, name, parent=nothing) where {T<:AccountType}
-        new{T}(account, code, name, parent, Vector{AccountInfo}(), Vector{AccountGroup}())
+    isdebit::Bool
+    parent::Union{Nothing,AccountGroup{B}}
+    subaccounts::StructArray{AccountInfo{B}}
+    subgroups::StructArray{AccountGroup{B}}
+
+    function AccountGroup{B}(
+        id,
+        code,
+        name,
+        isdebit=true,
+        parent=nothing,
+        subaccounts=StructArray(Vector{AccountInfo{B}}()),
+        subgroups=StructArray(Vector{AccountGroup{B}}())) where {B <: Position}
+        acc = new{B}(id, code, name, isdebit, parent, subaccounts, subgroups)
+        parent === nothing || push!(parent.subgroups, acc)
+        acc
     end
 end
 
+AccountGroup(
```

Comment:
if it's not one (at most two lines), use the long form, i.e. `function AccountGroup`